### PR TITLE
#166187807 Redux Setup

### PR DESCRIPTION
**Pull request by Emmanuelaaron**
#### What does this PR do?
* Setting redux and appropriate file structure
#### Description of task completed
* Set redux and an appropriate file structure
#### How should this be manually tested on your local machine
* Do a `git fetch origin ch-redux-setup-166187807`
* Do a `git checkout ch-redux-setup-166187807`
* Install all the dependencies using `npm install`
* install [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) in chrome
* Run the app using `npm start`
* Right-click on your browser and inspect.
* Click on `redux >> raw`
* You should see an empty state object

[Maintains #166187807](https://www.pivotaltracker.com/n/projects/2350340/stories/166187807)

### CODE REVIEW

File: src/store/index.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createStore, applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk';
+import reducers from '../reducer/index';
+
+const enhacers = compose(
+  applyMiddleware(thunk),
```
Comment (jennizalwango) [inline src/store/index.jsx:6]: Good work @Emmanuelaaron, 
Please help me understand what thunk does?
Thanks

Comment (alexxsanya) [reply src/store/index.jsx:6]: @jennizalwango you may visit this link https://medium.com/fullstack-academy/thunks-in-redux-the-basics-85e538a3fe60 to understand what **thunk** does.

**Comment by alexxsanya** [comment]
> #### What does this PR do?
> * Setting redux and appropriate file structure
> 
> #### Description of task completed
> * Set redux and an appropriate file structure
> 
> #### How should this be manually tested on your local machine
> * Do a `git fetch origin ch-redux-setup-16618780`
> * Do a `git checkout ch-redux-setup-16618780`
> * Install all the dependencies using `npm install`
> * install [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) in chrome
> * Run the app using `npm start`
> * Right-click on your browser and inspect.
> * Click on `redux >> raw`
> * You should see an empty state object
> 
> [Maintains #166187807](https://www.pivotaltracker.com/n/projects/2350340/stories/166187807)

@Emmanuelaaron the branch name you are using in the git commands are incorrect resulting into an error when you copy the command and use it. they are missing a 7 at the end. kindly update.

**Review by alexxsanya — CHANGES_REQUESTED** [review]
Screenshot 2019-07-06 at 15 46 37

@Emmanuelaaron above is what I see. I think that's not enough to confirm whether redux is setup. I suggest you consider adding a atleast a **state** to the store and we look out for that to confirm that things are setup well.

File: src/store/index.jsx (hunk 1 above)
Comment (Emmanuelaaron) [reply src/store/index.jsx:6]: @jennizalwango, Redux -thunk is a middleware for redux that allows us to directly access the dispatch method.

**Comment by Emmanuelaaron** [comment]
> > #### What does this PR do?
> > 
> > * Setting redux and appropriate file structure
> > 
> > #### Description of task completed
> > 
> > * Set redux and an appropriate file structure
> > 
> > #### How should this be manually tested on your local machine
> > 
> > * Do a `git fetch origin ch-redux-setup-16618780`
> > * Do a `git checkout ch-redux-setup-16618780`
> > * Install all the dependencies using `npm install`
> > * install [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) in chrome
> > * Run the app using `npm start`
> > * Right-click on your browser and inspect.
> > * Click on `redux >> raw`
> > * You should see an empty state object
> > 
> > [Maintains #166187807](https://www.pivotaltracker.com/n/projects/2350340/stories/166187807)
> 
> @Emmanuelaaron the branch name you are using in the git commands are incorrect resulting into an error when you copy the command and use it. they are missing a 7 at the end. kindly update.

Thanks @alexxsanya for taking note of that.
I have rectified it

**Review by neelxie — APPROVED** [review]
I have run the app and the redux works fine if I add my own custom state to the available component, good work on setting it up.

**Comment by Emmanuelaaron** [comment]
> Screenshot 2019-07-06 at 15 46 37
> 
> @Emmanuelaaron above is what I see. I think that's not enough to confirm whether redux is setup. I suggest you consider adding a atleast a **state** to the store and we look out for that to confirm that things are setup well.

Thanks @alexxsanya,
If you are to click on `raw`, you should be able to see an empty object. That's enough to confirm that redux is setup. Please refer to the screenshot below

Screenshot 2019-07-08 at 10 23 05

File: src/store/index.jsx (hunk 1 above)
Comment (jennizalwango) [reply src/store/index.jsx:6]: okay thanks, @Emmanuelaaron,
I think now you are good to go

Comment (HabibSentongo) [reply src/store/index.jsx:6]: Thank you all, I also didn't know what it does.

**Review by HabibSentongo — APPROVED** [review]
I have gone through all the changed files and I see you added all the required changes for redux to work, and the branch runs well on my local machine. So, I give this a green light.

**Comment by alexxsanya** [comment]
> > Screenshot 2019-07-06 at 15 46 37
> > @Emmanuelaaron above is what I see. I think that's not enough to confirm whether redux is setup. I suggest you consider adding a atleast a **state** to the store and we look out for that to confirm that things are setup well.
> 
> Thanks @alexxsanya,
> If you are to click on `raw`, you should be able to see an empty object. That's enough to confirm that redux is setup. Please refer to the screenshot below
> 
> Screenshot 2019-07-08 at 10 23 05

okay.

**Review by alexxsanya — APPROVED** [review]

**Review by jennizalwango — APPROVED** [review]

**Pull request merged**